planner/core/exhaust_physical_plans.go

```diff
@@ -1318,8 +1391,8 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) []Ph
 	}
 	joins = append(joins, indexJoins...)
 
-	hashJoins := p.getHashJoins(prop)
-	if (p.preferJoinType & preferHashJoin) > 0 {
+	hashJoins, forced := p.tryToGetHashJoins(prop)
+	if (p.preferJoinType&preferHashJoin) > 0 || forced {
```

Comment:
We may need treat `p.preferJoinType&preferHashJoin` as forced in `tryToGetHashJoins`.

Reply:
move it into `tryToGetHashJoins`